### PR TITLE
Avoid masking of difficulty adjustment errors by checkpoints

### PR DESCRIPTION
**Pull request by sipa**
Currently difficulty adjustment violations are not reported for chains that branch off before the last checkpoint. Change this by moving the checkpoint check after the difficulty check.

### CODE REVIEW

**Review by gmaxwell — APPROVED** [review]
ACK

**Comment by sdaftuar** [comment]
ACK

File: src/validation.cpp [hunk 1]
```diff
@@ -2830,22 +2830,6 @@ bool CheckBlock(const CBlock& block, CValidationState& state, const Consensus::P
     return true;
 }
 
-static bool CheckIndexAgainstCheckpoint(const CBlockIndex* pindexPrev, CValidationState& state, const CChainParams& chainparams, const uint256& hash)
```
Comment (promag) [inline src/validation.cpp]: Could keep even though it's called once?

**Review by promag — COMMENTED** [review]
utACK 85c82b5.

**Comment by laanwj** [comment]
utACK 85c82b5

**Pull request merged**